### ISSUE TITLE
`"FusedAdam"` and `"Adam"` optimizer types lead to different handling of weight decay

### ISSUE THREAD

**Issue by spencerkclark** [ai2cm/ace#661]
In reproducing and discussing unusual results shared with us by an external user, @mcgibbon, @Arcomano1234, and I noted that while our optimization configuration code accepts two possible values for the `optimizer_type`, `"FusedAdam"` or `"Adam"`, these differ in more than just the value of `fused` passed to the torch optimizer constructor.  They also differ in the `Optimizer` class used: https://github.com/ai2cm/full-model/blob/a6bf5da20f955f7fba7329702ac2473272d1d7b4/fme/core/optimization.py#L89-L94

with `"FusedAdam"` using `torch.optim.AdamW` and `"Adam"` using `torch.optim.Adam`. 

`torch.optim.AdamW` is a subclass of `torch.optim.Adam` with a different handling a weight decay, and therefore leads to different training behavior when weight decay is turned on.  For example, the 4° SHiELD-AMIP training jobs illustrated in this `batch_loss` plot were configured identically with the exception that one used the `"FusedAdam"` `optimizer_type` and the other used the `"Adam"` `optimizer_type` ([WandB](https://wandb.ai/ai2cm/ace/workspace/panel/wsh9itw1c?nw=nwuserspencerc_ai2)).  We should probably harmonize the `Optimizer` class used and/or modify the option names to make the distinction clearer.

<img width="2528" height="1328" alt="Image" src="https://github.com/user-attachments/assets/09618209-aed1-4a29-b4b1-9aba97ad0d2a" />
